accounts/abi/bind/backends/blockchain.go

```diff
@@ -141,3 +144,197 @@ func (b *BlockchainContractCaller) getBlockAndState(num *big.Int) (*types.Block,
 	state, err := b.bc.StateAt(block.Root())
 	return block, state, err
 }
+
+type BlockchainContractTransactor struct {
+	bc     BlockChainForCaller
+	txPool *blockchain.TxPool
+}
+
+// This nil assignment ensures at compile time that BlockchainContractCaller implements bind.ContractCaller.
+var _ bind.ContractTransactor = (*BlockchainContractTransactor)(nil)
+
+func NewBlockchainContractTransactor(bc BlockChainForCaller, txPool *blockchain.TxPool) *BlockchainContractTransactor {
+	return &BlockchainContractTransactor{bc, txPool}
+}
+
+func (b *BlockchainContractTransactor) PendingCodeAt(ctx context.Context, account common.Address) ([]byte, error) {
+	// TODO-Klaytn this is not pending code but latest code
+	state, err := b.bc.State()
+	if err != nil {
+		return nil, err
+	}
+	return state.GetCode(account), nil
+}
+
+func (b *BlockchainContractTransactor) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	if b.txPool != nil {
+		return b.txPool.GetPendingNonce(account), nil
+	}
+	// TODO-Klaytn this is not pending nonce but latest nonce
+	state, err := b.bc.State()
+	if err != nil {
+		return 0, err
+	}
+	return state.GetNonce(account), nil
+}
+
+func (b *BlockchainContractTransactor) SuggestGasPrice(ctx context.Context) (*big.Int, error) {
+	if b.bc.Config().IsMagmaForkEnabled(b.bc.CurrentHeader().Number) {
+		return new(big.Int).SetUint64(b.bc.Config().Governance.KIP71.UpperBoundBaseFee), nil
+	} else {
+		return new(big.Int).SetUint64(b.bc.Config().UnitPrice), nil
+	}
+}
+
+func (b *BlockchainContractTransactor) EstimateGas(ctx context.Context, call klaytn.CallMsg) (uint64, error) {
+	state, err := b.bc.State()
+	if err != nil {
+		return 0, err
+	}
+	balance := state.GetBalance(call.From) // from can't be nil
+
+	// Create a helper to check if a gas allowance results in an executable transaction
+	executable := func(gas uint64) (bool, *blockchain.ExecutionResult, error) {
+		call.Gas = gas
+
+		currentState, err := b.bc.State()
+		if err != nil {
+			return true, nil, nil
+		}
+		bc := NewBlockchainContractCaller(b.bc)
+		res, err := bc.callContract(call, b.bc.CurrentBlock(), currentState)
```

Comment:
Why didn't BlockchainContractTransactor inherit (i.e. embed) the BlockchainContractCaller?

In other words, is this possible?
```suggestion
		res, err := b.CallContract(context.Background(), call, nil)
```

Reply:
Yes, however we still need to call `callContract` (not `CallContract`) to get `*blockchain.ExecutionResult`..

Do you think it is more appropriate to embed `BlockchainContractCaller` in `BlockchainContractTransactor`? I implemented corresponding structs for `bind.Contract*` and there was no reason for not embedding `BlockchainContractCaller` in `BlockchainContractTransactor`. But I just found out that `bind.ContractTransactor` states that it `defines the methods needed to allow operating with contract on a write only basis` in this [comment](https://github.com/klaytn/klaytn/blob/v1.11.0/accounts/abi/bind/backend.go#L69).

Reply:
@2dvorak Sure thing.